src/Support/ClassLoader.php

```diff
@@ -135,28 +114,35 @@ class_alias($class, $reverse);
             return true;
         }
 
-        list($lowerClass, $upperClass, $lowerClassStudlyFile, $upperClassStudlyFile) = static::getPathsForClass($class);
-
-        foreach ($this->directories as $directory) {
-            $paths = [
-                $directory . DIRECTORY_SEPARATOR . $lowerClass,
-                $directory . DIRECTORY_SEPARATOR . $upperClass,
-                $directory . DIRECTORY_SEPARATOR . $lowerClassStudlyFile,
-                $directory . DIRECTORY_SEPARATOR . $upperClassStudlyFile,
-            ];
-
-            foreach ($paths as $path) {
-                if ($this->isRealFilePath($path)) {
-                    $this->includeClass($class, $path);
-
-                    if (!is_null($reverse = $this->getReverseAlias($class))) {
-                        if (!class_exists($reverse, false) && !in_array($reverse, $this->loadedAliases)) {
-                            class_alias($class, $reverse);
-                            $this->reversedClasses[] = $reverse;
+        // Check our registered autoload packages for a match
+        foreach ($this->autoloadedPackages as $prefix => $path) {
+            if (Str::startsWith($class, $prefix)) {
+                $parts = explode('\\', Str::after($class, $prefix));
+                $file = array_pop($parts) . '.php';
+                $namespace = implode('\\', $parts);
+                $directory = str_replace(['\\', '_'], DIRECTORY_SEPARATOR, $namespace);
+
+                $pathsToTry = [
+                    // Lowercase directory structure - default structure of plugins and modules
+                    $path . strtolower($directory) . DIRECTORY_SEPARATOR . $file,
+
+                    // Fallback to the unmodified path
+                    $path . $directory . DIRECTORY_SEPARATOR . $file,
+                ];
+
+                foreach ($pathsToTry as $classPath) {
+                    if ($this->isRealFilePath($classPath)) {
+                        $this->includeClass($class, $classPath);
+
+                        if (!is_null($reverse = $this->getReverseAlias($class))) {
```

Comment:
I'd prefer we define variables outside of conditional statements, so they can be analysed by Xdebug before going into the condition.
```suggestion
						$reverse = $this->getReverseAlias($class);
                        if (!is_null($reverse)) {
```